### PR TITLE
ht22 - new components (NotFound, Loader, Pages), routing Navigation c…

### PR DESCRIPTION
**Pull request by CherpakovPhilipp**
…omp.

### CODE REVIEW

File: src/components/navigation/Navigation.js [hunk 1]
```diff
@@ -4,17 +4,66 @@ import { Link, NavLink } from 'react-router-dom';
...
+  :
+    links.filter(item => {
+      if (item.auth === false || item.auth === undefined) {
+        return (
+          <li key={item.label}>
+            <NavLink to={`/${item.path}`}>{item.label}</NavLink>
+          </li>
+        )
+      }
+    })
```
Comment (iliatcymbal) [inline src/components/navigation/Navigation.js]: 1) lots of repeated code and the only differences are 
```
if (item.auth === true
if (item.auth === false
```
2) please, check how Array.prototype.filter works (http://learn.javascript.ru/array-iteration#filter)

File: src/pages/greeting/Greeting.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  else dayTime = 'night';
+
+  const greetNonAuth = (
+    <>
+      <h1>Welcome to the Toy market</h1>
+      <p>We are pleased to announce the launch of our brand new website!</p>
+      <p>From the 1st December our company starts to deliver online order.</p>
+      <p>First hundred customers gets a 10% discount!</p>
+    </>
+  );
```
Comment (iliatcymbal) [inline src/pages/greeting/Greeting.js]: Be more creative, don't copypaste content

File: src/pages/greeting/Greeting.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import { Infobar } from '../../components/infoBar';
+
+export const Greeting = ({ user }) => {
+  const time = new Date().getHours();
+  let dayTime;
+
+  if (time >= 3 && time <= 12) dayTime = 'morning';
+  else if (time > 12 && time < 18) dayTime = 'afternoon';
+  else if (time >= 18 && time <= 22) dayTime = 'evening';
+  else dayTime = 'night';
```
Comment (iliatcymbal) [inline src/pages/greeting/Greeting.js]: do you need that code?

File: src/pages/products/Products.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import './products.scss';
+import { Link } from 'react-router-dom';
+
+const products = ["Bicycle", "Car", "Super-hero", "Doll", "Teddy", "Lego", "Pistols", "Table-game"];
+
+export class Products extends Component {
+  state = {
+    products: products,
```
Comment (iliatcymbal) [inline src/pages/products/Products.js]: `: products` no need

File: src/pages/products/Products.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        this.products = data;
+        this.setState({ products: data });
+      });
+  }
+
+  setInputVal = ({ target }) => {
+    this.setState({ inputVal: target.value });
+  }
+
+  filterProducts = product => product.includes(this.state.inputVal)
```
Comment (iliatcymbal) [inline src/pages/products/Products.js]: make case insensitive filtering

File: src/pages/greeting/Greeting.js (hunk 2 above)
Comment (CherpakovPhilipp) [reply src/pages/greeting/Greeting.js]: It was temporary)

File: src/pages/greeting/Greeting.js (hunk 3 above)
Comment (CherpakovPhilipp) [reply src/pages/greeting/Greeting.js]: yes, to detect the time of day

Comment (CherpakovPhilipp) [reply src/pages/greeting/Greeting.js]: or, maybe, is here something wrong?

Comment (iliatcymbal) [reply src/pages/greeting/Greeting.js]: I mean I don't remember that component in the project design

**Pull request merged**